backend/src/services/category.go

```diff
@@ -12,19 +12,69 @@ import (
 )
 
 type CategoryServiceInterface interface {
-	CreateCategory(category models.Category) (*models.Category, error)
-}
+	CreateCategory(category models.CreateUpdateCategoryRequestBody) (*models.Category, error)
+	GetCategories() (*[]models.Category, error)
```

Comment:
```suggestion
	GetCategories() ([]models.Category, error)
```
for consistency, let's return an empty slice instead of a nil pointer. Agreed @DOOduneye ?